### PR TITLE
 [Feature] Marquee Text Control

### PR DESCRIPTION
**Pull request by Avid29**
Adds a new MarqueeText control that scrolls text across the screen as a Marquee.

https://user-images.githubusercontent.com/9384045/190292642-e187f29e-8024-4906-bcb0-368544dafc85.mp4

### CODE REVIEW

**Comment by Avid29** [comment]
Sorry for the resubmit. I realized the base branch name was wrong

File: labs/MarqueeText/src/MarqueeText.xaml
Comment (niels9001) [inline labs/MarqueeText/src/MarqueeText.xaml:90]:
```suggestion
                                           win:FontStretch="{TemplateBinding FontStretch}"
```

File: labs/MarqueeText/src/MarqueeText.xaml
Comment (niels9001) [inline labs/MarqueeText/src/MarqueeText.xaml:80]:
```suggestion
                                           win:FontStretch="{TemplateBinding FontStretch}"
```

File: labs/MarqueeText/src/MarqueeText.xaml
Comment (niels9001) [inline labs/MarqueeText/src/MarqueeText.xaml:3]:
```suggestion
                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                    xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
```

**Comment by niels9001** [comment]
@Avid29 Awesome work :)! FYI, I've added the win: prefix for FontStretch (as it's not supported for Uno) so that the build will hopefully pass!

**Comment by Avid29** [comment]
So the current issue appears to be that in Uno (and only in Uno) ``Timeline`` is ``IDisposible``, and therefore the ``MarqueeText`` control must also be ``IDisposible`` since it contains a ``Storyboard``. However in UWP there's no reason to make the control ``IDisposible``, so I'm really not tempted to just add that interface for no reason. For now, I have conditionally attached a ``IDisposible`` for Uno, but I'm not too happy with this solution.
image


Any better ideas?

**Comment by michael-hawker** [comment]
@Avid29 was there a warning message about this or something somewhere calling this out?

@jeromelaban is this `IDisposable` message a red herring, are we missing something else instead?

**Comment by jeromelaban** [comment]
Some objects are marked as disposable for internal constraints, but it's not something we can change right away as it may be a breaking change (we'll take note to add it as such for our next major). Having to add `IDisposable` here is likely a result of some analyzer that can be disabled for Uno targets specifically.

**Comment by Avid29** [comment]
Hmm. That's frustrating. It fails with ``IDisposable`` because it's redundant, and it fails without ``IDisposible`` because it's missing...

**Comment by Arlodotexe** [comment]
> Having to add IDisposable here is likely a result of some analyzer that can be disabled for Uno targets specifically.

@Avid29 See if you can find which analyzer is doing that. Since `IDisposable` is implemented as an internal constraint to Uno, we can effectively ignore this and disable the analyzer on uno platforms.

**Comment by Arlodotexe** [comment]
> > Having to add IDisposable here is likely a result of some analyzer that can be disabled for Uno targets specifically.
> 
> @Avid29 See if you can find which analyzer is doing that. Since `IDisposable` is implemented as an internal constraint to Uno, we can effectively ignore this and disable the analyzer on uno platforms.

Partially fixed in #377, but since MarqueeText still owns the field, we need to manually suppress the error in this file. There's no way to suppress this problem globally for all files.

**Review by Arlodotexe — APPROVED** [review]

**Comment by Avid29** [comment]
@niels9001 I believe this is just waiting on your approval, right @Arlodotexe?

File: components/MarqueeText/samples/MarqueeText.md
Comment (niels9001) [inline components/MarqueeText/samples/MarqueeText.md:18]: This can be removed, we have the discussion button that will take users there (using the discussion ID).

File: components/MarqueeText/samples/MarqueeText.md
Comment (niels9001) [inline components/MarqueeText/samples/MarqueeText.md:22]: Maybe mention the looping/speed properties?

**Comment by niels9001** [comment]
> @niels9001 I believe this is just waiting on your approval, right @Arlodotexe?

2 minor comments.. otherwise looking great 😁! Awesome control!

File: components/MarqueeText/samples/MarqueeText.md
Comment (niels9001) [inline components/MarqueeText/samples/MarqueeText.md:18]:
```suggestion
```

File: components/MarqueeText/samples/MarqueeText.md
Comment (niels9001) [inline components/MarqueeText/samples/MarqueeText.md:20]:
```suggestion
```

File: components/MarqueeText/samples/MarqueeText.md
Comment (niels9001) [inline components/MarqueeText/samples/MarqueeText.md:22]: Maybe mention the Behavior / Direction / Speed / RepeatBehavior here?

Comment (Avid29) [reply components/MarqueeText/samples/MarqueeText.md]: Yeah, I kind of ignored the markdown file. I just pushed a new commit with some more full docs, but I couldn't get multiple demo pages without the compiler blowing up 🤷‍♂️

Comment (niels9001) [reply components/MarqueeText/samples/MarqueeText.md]: Changes looking great :)! Thanks!

**Review by niels9001 — APPROVED** [review]

**Comment by Avid29** [comment]
In case anyone finds their way here, the `MarqueeText` control has been replaced by simply the `Marquee` control as of this PR:
https://github.com/CommunityToolkit/Labs-Windows/pull/732

**Pull request merged**
